includes/class-reader-activation.php

```diff
@@ -935,12 +936,23 @@ public static function process_auth_form() {
 			return;
 		}
 		$action   = isset( $_POST['action'] ) ? \sanitize_text_field( $_POST['action'] ) : '';
-		$email    = isset( $_POST['email'] ) ? \sanitize_email( $_POST['email'] ) : '';
+		$email    = isset( $_POST['npe'] ) ? \sanitize_email( $_POST['npe'] ) : '';
 		$password = isset( $_POST['password'] ) ? \sanitize_text_field( $_POST['password'] ) : '';
 		$redirect = isset( $_POST['redirect'] ) ? \esc_url_raw( $_POST['redirect'] ) : '';
 		$lists    = isset( $_POST['lists'] ) ? array_map( 'sanitize_text_field', $_POST['lists'] ) : [];
+		$honeypot = isset( $_POST['email'] ) ? \sanitize_email( $_POST['email'] ) : '';
 		// phpcs:enable
 
+		// Honeypot trap.
+		if ( ! empty( $honeypot ) ) {
+			return self::send_auth_form_response(
+				[
+					'email'         => $honeypot,
+					'authenticated' => \wp_rand( 1000000, PHP_INT_MAX ),
```

Comment:
The success result here would be bool (0/1). Why are we sending a random int?

Reply:
I misinterpreted the type of the `authenticated` value—I thought it was supposed to be a user ID. Fixed in ae7690d.